### PR TITLE
Advice on IThemeSelector

### PR DESCRIPTION
**Pull request by zhhbo**
I think we can modify the definition of IThemeSelector as follows,
 public interface IThemeSelector
    {
        /// <summary>
        /// Returns a <see cref="ThemeSelectorResult"/> representing a specific theme
        /// and a priority order. The highest priority value will be used.
        /// </summary>
        /// <returns></returns>
        Task<ThemeSelectorResult> GetThemeAsync();
        /// <summary>
        /// if CanSet is true,this will be used to set the theme
        /// </summary>
        /// <param name="themeName"></param>
        void SetTheme(string themeName);
        /// <summary>
        /// it is a flag to tell the stysem,the selector either can be seted or not.
        /// </summary>
        bool CanSet { get; }
        /// <summary>
        /// this will tell the system,the theme.txt need contains this Tag
        /// </summary>
        string Tag { get; }
        /// <summary>
        /// a Display name,that will be used in Orchard.Themes.AdminMenu to build a menu
        /// </summary>
        LocalizedString DisplayName { get; }
        /// <summary>
        /// this like a parameter to build a menu
        /// </summary>
        string Name { get; }
    }
This enables the implementation of the required template for different scenarios
For example, we can set up a website template, the Admin template, or you can also set the mobile terminal templates, etc.
it will be like this:
settheme

### CODE REVIEW

**Comment by zhhbo** [comment]
Click setSite maybe like this:
setsite
Click setAdmin maybe like this:
setadmin

**Comment by zhhbo** [comment]
and if do this ,we must modify AdminThemeSelector  like this:
 public class AdminThemeSelector : IThemeSelector
    {
        private readonly IAdminThemeService _adminThemeService;
        private readonly IHttpContextAccessor _httpContextAccessor;

        public AdminThemeSelector(
            IStringLocalizer<AdminThemeSelector>s,
            IAdminThemeService adminThemeService,
            IHttpContextAccessor httpContextAccessor
            )
        {
            _adminThemeService = adminThemeService;
            _httpContextAccessor = httpContextAccessor;
            T = s;
        }

        public async Task<ThemeSelectorResult> GetThemeAsync()
        {
            if (AdminAttribute.IsApplied(_httpContextAccessor.HttpContext))
            {
                string adminThemeName = await _adminThemeService.GetAdminThemeNameAsync();
                if (String.IsNullOrEmpty(adminThemeName))
                {
                    return null;
                }

                return new ThemeSelectorResult
                {
                    Priority = 100,
                    ThemeName = adminThemeName
                };
            }

            return null;
        }
        IStringLocalizer T { get; set; }
        public bool CanSet { get { return true; } }
        public void SetTheme(string themeName)
        {
            _adminThemeService.SetAdminThemeAsync(themeName);
        }

        public string Tag { get { return "admin"; } }

        public LocalizedString DisplayName { get { return T["AdminTheme"]; } }

        public string Name { get { return "adminTheme"; } }
    }

and so on.

**Comment by zhhbo** [comment]
Hope to be adopted and I will also modify the implementation of O1

**Comment by zhhbo** [comment]
I have copied some of the antoine and sipke code, I am very sorry about this point, because I am a new git user, will not be applied to merge, pull, etc., can only manually copy some of the necessary code.

**Comment by sebastienros** [comment]
We will definitely merge their PR first before accepting your work.

Also, as a general rule please file an issue with your suggestions, they look interesting but we'd like to be able to talk about it before having you start a huge piece work.

File: src/OrchardCore/Orchard.Environment.Extensions/Manifests/ManifestInfo.cs [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
+        {
+            var value = _configurationRoot["version"];
+
+            if (string.IsNullOrWhiteSpace(value))
+                return new Version(0, 0);
+
+            Version version;
+            Version.TryParse(value, out version);
+            return version;
+        }
```
Comment (Jetski5822) [inline src/OrchardCore/Orchard.Environment.Extensions/Manifests/ManifestInfo.cs]: Hmmmm hang on... I need to think about this, I have been trying not to introduce these in here.

**Comment by sebastienros** [comment]
Closing the PR because there are too many conflicts, but we are keeping the issue targetting RC as it's a good suggestion. We'll revisit it soon.

**Pull request closed without merging**